server/src/test/java/org/elasticsearch/search/aggregations/bucket/DateHistogramIT.java

```diff
@@ -1393,15 +1398,19 @@ public void testRewriteTimeZone_EpochMillisFormat() throws InterruptedException,
         indexRandom(true, client().prepareIndex(index, "type").setSource("d", "1477954800000"));
         ensureSearchable(index);
         SearchResponse response = client().prepareSearch(index).addAggregation(dateHistogram("histo").field("d")
-                .dateHistogramInterval(DateHistogramInterval.MONTH).timeZone(DateTimeZone.forID("Europe/Berlin"))).get();
+                .dateHistogramInterval(DateHistogramInterval.MONTH).timeZone(ZoneId.of("Europe/Berlin"))).get();
         assertSearchResponse(response);
         Histogram histo = response.getAggregations().get("histo");
         assertThat(histo.getBuckets().size(), equalTo(1));
-        assertThat(histo.getBuckets().get(0).getKeyAsString(), equalTo("1477954800000"));
+        if (JavaVersion.current().getVersion().get(0) == 8 && histo.getBuckets().get(0).getKeyAsString().endsWith(".0")) {
```

Comment:
I wonder whether this behavior causes problems on Java upgrades and whether it makes sense that we at least document it?

Reply:
absolutely, those java8/11 differenes will be documented